### PR TITLE
MongoVersion: Check BestAPIVersion before making a call

### PR DESCRIPTION
**Pull request by SimonRichardson**
## Description of change

The following ensures that we check if we can even make the call
before calling it. Also hide the mongo version if it's empty, cleaning
up the output from the API.

## QA steps

Bootstap 2.5 branch
```
juju bootstrap lxd old
```

Upgrade to juju develop/edge
```
juju show-controller
```

### CODE REVIEW

**Comment by SimonRichardson** [comment]
https://trello.com/c/LLYpkOMy/1076-ensure-mongo-version-works-with-older-controllers

File: cmd/juju/controller/showcontroller.go [hunk 1]
```diff
@@ -168,6 +169,23 @@ func (c *showControllerCommand) Run(ctx *cmd.Context) error {
...
+		if apiVersion := client.BestAPIVersion(); apiVersion >= 6 {
+			mongoVersion, err = client.MongoVersion()
+			if err != nil {
+				details.Errors = append(details.Errors, err.Error())
+				continue
+			}
+		}
+
+		// Update client store.
+		if err := c.SetControllerModels(c.store, controllerName, allModels); err != nil {
```
Comment (achilleasa) [inline cmd/juju/controller/showcontroller.go]: I think this block should be removed; it fails with a permission error if the user has no SuperuserAccess so I had that moved to L164 to keep things working for users with login (or with the add-model) permission.

Comment (SimonRichardson) [reply cmd/juju/controller/showcontroller.go]: You're right it's a rebase issue.

**Comment by achilleasa** [comment]
LGTM. My only concern is that the minimum API version for the call is hardcoded at the call site **and** inside the api -> facade [bridge](https://github.com/juju/juju/blob/develop/api/controller/mongo.go) code which could prove to be problematic if we plan on calling this from different locations.

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**